### PR TITLE
feature: Allow Docker Compose volumes to be kept after IT complete

### PR DESCRIPTION
**Pull request by hsiliev**
Adds a script that can configure environment variable controlling this behaviour

Solves: PZ-3153

### CODE REVIEW

File: start-it-with-local-env.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+if [ $build = "true" ]; then
+  echo "Building ZAC Docker Image ..."
+  ./gradlew buildDockerImage
+fi
+
+export ZAC_DOCKER_IMAGE=ghcr.io/infonl/zaakafhandelcomponent:dev
+[ -f check-env.sh ] && ./check-env.sh
+
+# shellcheck disable=SC2086
```
Comment (edgarvonk) [inline start-it-with-local-env.sh:65]: what is this?

File: .gitignore [hunk 2]
```diff
@@ -37,3 +37,5 @@ scripts/github/.secrets
 
 ### Docker Compose ###
 docker-compose.override.yml
+check-env.sh
```
Comment (edgarvonk) [inline .gitignore:40]: Why ignore these files?

File: scripts/docker-compose/check-env.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -e
+
```
Comment (edgarvonk) [inline scripts/docker-compose/check-env.sh:4]: Add some doc about what this script does? Not immediately clear to me. Maybe also rename it a bit to make it more clear?

File: scripts/docker-compose/check-env.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# SPDX-License-Identifier: EUPL-1.2+
+#
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+(
+  cd "$SCRIPT_DIR"
+
+  if sudo lsof -i@localhost:8080; then
+    echo -e "\nPlease stop the currently running WildFly!"
```
Comment (edgarvonk) [inline scripts/docker-compose/check-env.sh:16]: Can be anything running on port 8080. Does not need to be WildFly/ZAC. I would keep it neutral. Just say that there is process running on this port?

File: start-it-with-local-env.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+
+set -e
+
+help() {
+   echo "Starts the Integration Tests with local docker image."
```
Comment (edgarvonk) [inline start-it-with-local-env.sh:6]:
```suggestion
   echo "Starts the integration tests with a local ZAC Docker Image."
```

;-)

File: start-it-with-local-env.sh [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+   echo
+   echo "Syntax: $0 [-l|d|v|u|h]"
+   echo "options:"
+   echo "-l     Build a local ZAC Docker image"
+   echo "-d     Delete local Docker volume data before starting Docker Compose."
```
Comment (edgarvonk) [inline start-it-with-local-env.sh:11]:
```suggestion
   echo "-d     Delete local Docker volume data before starting Docker Compose"
```

File: start-it-with-local-env.sh [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+   echo "-v     Keep local Docker Compose volume data after test execution"
+   echo "-u     Turn on debug logs"
+   echo "-h     Print this Help."
```
Comment (edgarvonk) [inline start-it-with-local-env.sh:14]:
```suggestion
   echo "-h     Print this Help"
```

**Review by edgarvonk — CHANGES_REQUESTED** [review]
nice one!

File: start-it-with-local-env.sh (hunk 1 above)
Comment (hsiliev) [reply start-it-with-local-env.sh:65]: shellcheck wants to have quoted arguments in the next line, but this breaks the Gradle parsing, so I just ignore this warning

https://www.shellcheck.net/wiki/SC2086

File: .gitignore (hunk 2 above)
Comment (hsiliev) [reply .gitignore:40]: These are only needed for Docker daemon running as root (Linux systems)

File: scripts/docker-compose/check-env.sh (hunk 3 above)
Comment (hsiliev) [reply scripts/docker-compose/check-env.sh]: Renamed to check-for-running-containers.sh

**Review by edgarvonk — APPROVED** [review]

**Pull request merged**